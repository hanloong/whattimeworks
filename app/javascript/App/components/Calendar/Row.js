import React, { Component } from "react";
import { TableRowColumn } from "material-ui/Table";
import * as colors from "material-ui/styles/colors";

const style = {
  valid: {
    backgroundColor: colors.teal100
  },
  almost: {
    backgroundColor: colors.teal50
  },
  invalid: {
    backgroundColor: colors.grey50,
    color: colors.grey400
  }
};

class Row extends Component {
  getDiffDays({ days }) {
    if (days === 0) {
      return "";
    }

    return days > 0 ? `(+${days})` : `(-${days})`;
  }

  cellStyle({ status }) {
    switch (status) {
      case "VALID":
        return style.valid;
      case "ALMOST":
        return style.almost;
      default:
        return style.invalid;
    }
  }

  render() {
    const { row } = this.props;
    const cells = row.times.map((time, i) => {
      return (
        <TableRowColumn key={i} style={this.cellStyle(time)}>
          <span style={time.past ? { textDecoration: "line-through" } : {}}>
            {time.startTime} - {time.endTime}
            {this.getDiffDays(time)}
          </span>
        </TableRowColumn>
      );
    });

    return (
      <tr>
        {cells}
        <TableRowColumn />
      </tr>
    );
  }
}

export default Row;
