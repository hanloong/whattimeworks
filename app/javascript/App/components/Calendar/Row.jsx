import React from 'react'
import Cell from './Cell'
import { TableRow, TableRowColumn } from 'material-ui/Table';

const Row = ({ row }) => {
  return <TableRow>
    {
      row.times.map((time, i) =>
        <Cell
          valid={ row.valid }
          key={ i }
          time={ time } />
      )
    }
    <TableRowColumn/>
  </TableRow>
}

module.exports = Row
