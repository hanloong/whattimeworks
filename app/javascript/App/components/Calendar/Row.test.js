import React from 'react';
import Row from './Row';
import renderer from 'react-test-renderer';

test('Print out a table cell', () => {
  const row = {
    times: [{
      time: {
        time: 'Today, 3pm',
        zone: 'Australia/Sydney'
      }
    }],
    valid: false
  }

  const component = renderer.create(
    <Row row={ row } />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
