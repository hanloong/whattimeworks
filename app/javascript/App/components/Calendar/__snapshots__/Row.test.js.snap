// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print out a table row 1`] = `
<tr
  className={undefined}
  style={
    Object {
      "borderBottom": "1px solid #e0e0e0",
      "color": "rgba(0, 0, 0, 0.87)",
      "height": 48,
      "muiPrepared": true,
    }
  }
>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    Mon, 12:00 AM
  </td>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  />
</tr>
`;
